### PR TITLE
Fix the javadoc warnings, #166

### LINKED ISSUE #166: Investigate javadoc WARNINGS when executing `mvn javadoc:javadoc`
This is a follow up to #164 

The current set of warnings to investigate are:

```
9 warnings
[WARNING] Javadoc Warnings
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\ConditionalTagSupport.java:55: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected abstract boolean condition() throws JspTagException;
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\IndexedValueExpression.java:37: warning: no @param for orig
[WARNING] public IndexedValueExpression(ValueExpression orig, int i) {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\IndexedValueExpression.java:37: warning: no @param for i
[WARNING] public IndexedValueExpression(ValueExpression orig, int i) {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\IteratedExpression.java:55: warning: no @return
[WARNING] public Object getItem(ELContext context, int i) {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:185: warning: no description for @exception
[WARNING] * @exception jakarta.servlet.jsp.JspTagException
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:196: warning: no description for @exception
[WARNING] * @exception jakarta.servlet.jsp.JspTagException
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:453: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected void validateBegin() throws JspTagException {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:462: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected void validateEnd() throws JspTagException {
[WARNING] ^
[WARNING] C:\jakartaGit\jstl-api\api\src\main\java\jakarta\servlet\jsp\jstl\core\LoopTagSupport.java:471: warning: no @throws for jakarta.servlet.jsp.JspTagException
[WARNING] protected void validateStep() throws JspTagException {
[WARNING] ^
```

### PR DESCRIPTION
**Pull request by starksm64**
Signed-off-by: Scott M Stark <starksm64@gmail.com>

fixes #166

### CODE REVIEW

File: api/src/main/java/jakarta/servlet/jsp/jstl/core/IndexedValueExpression.java [hunk 1]
```diff
@@ -32,7 +32,8 @@
     protected final ValueExpression orig;
 
     /**
-     *
+     * @param orig - ValueExpression that refers to a specific member of an indexed variable
```
Comment (pzygielo) [inline api/src/main/java/jakarta/servlet/jsp/jstl/core/IndexedValueExpression.java:35]: Unnecessary `-`s.

**Review by volosied — APPROVED** [review]
You could remove the dashes as pzygielo commented.

Tested locally, and the warnings disappeared. (Thanks for point out about running install first before javadoc)

**Comment by starksm64** [comment]
Ok, I have removed the unnecessary dashes from javadoc comments

**Comment by pnicolucci** [comment]
I linked this PR to issue #166

File: api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java [hunk 2]
```diff
@@ -450,6 +450,7 @@ public void setVarStatus(String statusId) {
 
     /**
      * Ensures the "begin" property is sensible, throwing an exception expected to propagate up if it isn't
+     * @throws JspTagException - thrown on error
      */
```
Comment (pnicolucci) [inline api/src/main/java/jakarta/servlet/jsp/jstl/core/LoopTagSupport.java:454]: Looks like one `-` was missed in your removal commit.

**Review by pnicolucci — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**